### PR TITLE
fix: [CDS-41392]: move ClonePipeline modal to the top of the tree in the pipeline list instead of repeating for each item

### PR DESCRIPTION
**Pull request by udayvunnam**
### Summary

- This fixes bubbling events for row/card click that takes the user to pipeline studio without intent https://harness.atlassian.net/browse/CDS-41392
- Note: `src/modules/70-pipeline/pages/pipelines/views/PipelineGridView.tsx and src/modules/70-pipeline/pages/pipelines/views/PipelineListView` are temporary until we make a permanent switch to new URL/UX

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by udayvunnam** [comment]
retrigger titlecheck

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 14 Issues 
* Code issue 14 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/e864badd0b95ab81c060b15b85f15701/pull-request/10594)

**Review by vkbansal-harness — APPROVED** [review]

File: src/modules/70-pipeline/pages/pipelines/views/PipelineListView.tsx [hunk 1]
```diff
@@ -175,7 +163,8 @@ const RenderColumnMenu: Renderer<CellProps<PipelineDTO>> = ({ row, column }) =>
             text={getString('projectCard.clone')}
             onClick={(e: React.MouseEvent) => {
               e.stopPropagation()
-              openClonePipelineModal()
+              ;(column as any).onClonePipeline(data)
```
Comment (AshwinBhatkal-Harness) [inline src/modules/70-pipeline/pages/pipelines/views/PipelineListView.tsx:166]: nit: PipelineListViewProps instead of any

**Review by AshwinBhatkal-Harness — APPROVED** [review]

File: src/modules/70-pipeline/pages/pipelines/views/PipelineListView.tsx (hunk 1 above)
Comment (udayvunnam) [reply src/modules/70-pipeline/pages/pipelines/views/PipelineListView.tsx:166]: We can do `column as unknown as PipelineListViewProps` that's definitely better. I will change all instances of this for the table in the upcoming PR. Hope these types be fixed with ReactTable upgrade if that ever happens. This

**Pull request merged**